### PR TITLE
Add formal verification of mret in CI / CD

### PR DESCRIPTION
**Pull request by francois141**
Another test to verify the security monitor!

### CODE REVIEW

**Comment by francois141** [comment]
The first milestone for the automated formal verification is ready! @CharlyCst Can you please make sure the just install pipeline is still easy to use?

**Comment by CharlyCst** [comment]
Cool, it seems like the Ci was failing because Kani is too verbose ^^
Turns out (after some deeper digging) that there is a flag to tune the verbosity, that fixed the issue :)

Before merging this I want to fix a few things, I will come back to it soon.

File: src/arch/pmp.rs [hunk 1]
```diff
@@ -228,7 +228,11 @@ impl PmpGroup {
             permissions < 8,
             "Permissions should not set NAPOT or TOP bits"
         );
-        self.set(idx, build_napot(from, to).unwrap(), permissions | NAPOT);
+
+        match build_napot(from, to) {
+            Some(value) => self.set(idx, value, permissions | NAPOT),
+            None => panic!("Invalid arguments for build_napot"),
+        }
```
Comment (CharlyCst) [inline src/arch/pmp.rs:235]: Out of curiosity why did you need to change that one? It seems semantically equivalent, was Kani complaining about it?

Comment (francois141) [reply src/arch/pmp.rs:235]: They are equivalent, I believed Kani complained about that but it was because I was giving an hardcoded illegal value in the kani verify function. You can revert if you want
